apps/web/src/components/virtualized-tree/index.tsx

```diff
@@ -95,7 +99,7 @@ export function VirtualizedTree<T>(props: TreeViewProps<T>) {
     treeRef,
     () => ({
       async refresh() {
-        setNodes([]);
+        // setNodes([]);
```

Comment:
If we don't do this, the items duplicate during filter.

Reply:
I see.

When a notebook/subnotebook is created, both `refresh` and `refreshItem` is triggered. Nodes are reset in `refresh` in the above line, but it doesn't get populated again before `refreshItem` tries to access nodes. Which causes an error and crashes the app.

I don't fully understand why removing the above line causes notebooks to be duplicated while filtering. I've created a separate `resetAndRefresh` method for filtering notebook usecase which does fix the issue.

Reply:
> I don't fully understand why removing the above line causes notebooks to be duplicated while filtering.

It's something to do with virtuouso not removing old items from the list.

> I've created a separate resetAndRefresh method for filtering notebook usecase which does fix the issue.

Sounds good.